From f90a349979926bb547583dc6f1b6b1f1ab3b3189 Mon Sep 17 00:00:00 2001
From: liuchuanqi <legend050709@qq.com>
Date: Fri, 7 Aug 2020 19:20:57 +0800
Subject: [PATCH] rm rte_experimental attribute of rte_memseg_walk

there is no __rte_experimental attribute in function rte_mempool_walk and rte_memzone_walk of dpdk 18.11,
and there is no __rte_experimental attribute in function rte_memseg_walk of the higher version's dpdk(eg: dpdk 20.05).
so remove it to prevent compilation error when dpdk application calls the function.
---
 lib/librte_eal/common/eal_common_memory.c  | 2 +-
 lib/librte_eal/common/include/rte_memory.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/librte_eal/common/eal_common_memory.c b/lib/librte_eal/common/eal_common_memory.c
index d47ea4938..3d8ce67f0 100644
--- a/lib/librte_eal/common/eal_common_memory.c
+++ b/lib/librte_eal/common/eal_common_memory.c
@@ -601,7 +601,7 @@ rte_memseg_contig_walk(rte_memseg_contig_walk_t func, void *arg)
 	return ret;
 }
 
-int __rte_experimental
+int
 rte_memseg_walk_thread_unsafe(rte_memseg_walk_t func, void *arg)
 {
 	struct rte_mem_config *mcfg = rte_eal_get_configuration()->mem_config;
diff --git a/lib/librte_eal/common/include/rte_memory.h b/lib/librte_eal/common/include/rte_memory.h
index d970825df..71bee8b6b 100644
--- a/lib/librte_eal/common/include/rte_memory.h
+++ b/lib/librte_eal/common/include/rte_memory.h
@@ -227,7 +227,7 @@ typedef int (*rte_memseg_list_walk_t)(const struct rte_memseg_list *msl,
  *   1 if stopped by the user
  *   -1 if user function reported error
  */
-int __rte_experimental
+int
 rte_memseg_walk(rte_memseg_walk_t func, void *arg);
 
 /**
-- 
2.21.1 (Apple Git-122.3)

